=pro #16695 Update dependencies

patriknw opened the pull request:
io.netty:netty: 3.8.0.Final -> 3.8.3.Final
org.slf4j:slf4j-api: 1.7.7 -> 1.7.12
org.fusesource.leveldbjni:leveldbjni-all:optional;provided: 1.7 -> 1.8
org.apache.camel:camel-core: 2.13.0 -> 2.13.4

ch.qos.logback:logback-classic:test: 1.0.13 -> 1.1.3
junit:junit:test: 4.10 -> 4.12
org.scalatest:scalatest:test: 2.1.3 -> 2.1.7
org.mockito:mockito-all:test: 1.9.5 -> 1.10.19
org.scala-lang.modules:scala-xml:test: 1.0.1 -> 1.0.4
com.novocode:junit-interface:test : 0.8 -> 0.11
org.slf4j:jul-to-slf4j:test: 1.7.7  -> 1.7.12
org.slf4j:log4j-over-slf4j:test: 1.7.7  -> 1.7.12
com.codahale.metrics:metrics-core:test: 3.0.1  -> 3.0.2
com.codahale.metrics:metrics-jvm:test: 3.0.1  -> 3.0.2
org.scalacheck:scalacheck:test: 1.11.3 -> 1.11.6
org.apache.commons:commons-math:test: 2.1 -> 2.2
commons-codec:commons-codec:test: 1.7 -> 1.10

patriknw committed:
=pro #16695 Update dependencies

io.netty:netty: 3.8.0.Final -> 3.10.3.Final
org.slf4j:slf4j-api: 1.7.7 -> 1.7.12
org.fusesource.leveldbjni:leveldbjni-all:optional;provided: 1.7 -> 1.8
org.apache.camel:camel-core: 2.13.0 -> 2.13.4

ch.qos.logback:logback-classic:test: 1.0.13 -> 1.1.3
junit:junit:test: 4.10 -> 4.12
org.scalatest:scalatest:test: 2.1.3 -> 2.2.4
org.mockito:mockito-all:test: 1.9.5 -> 1.10.19
org.scala-lang.modules:scala-xml:test: 1.0.1 -> 1.0.4
com.novocode:junit-interface:test : 0.8 -> 0.11
org.slf4j:jul-to-slf4j:test: 1.7.7  -> 1.7.12
org.slf4j:log4j-over-slf4j:test: 1.7.7  -> 1.7.12
com.codahale.metrics:metrics-core:test: 3.0.1  -> 3.0.2
com.codahale.metrics:metrics-jvm:test: 3.0.1  -> 3.0.2
org.scalacheck:scalacheck:test: 1.11.3 -> 1.11.6
org.apache.commons:commons-math:test: 2.1 -> 2.2
commons-codec:commons-codec:test: 1.7 -> 1.10

drewhk commented:
Wow, good job! LGTM!

ktoso commented on project/Dependencies.scala:
```diff
@@ -11,23 +11,23 @@ object Dependencies {
     val scalaStmVersion  = sys.props.get("akka.build.scalaStmVersion").getOrElse("0.7")
-    val scalaTestVersion = sys.props.get("akka.build.scalaTestVersion").getOrElse("2.1.3")
-    val scalaCheckVersion = sys.props.get("akka.build.scalaCheckVersion").getOrElse("1.11.3")
+    val scalaTestVersion = sys.props.get("akka.build.scalaTestVersion").getOrElse("2.1.7")
```
We'll even hop on `2.2.4` with just one minor change (I have that in my PR) :-)
// Feel free to merge this one first.

ktoso commented on project/plugins.sbt:
```diff
@@ -29,3 +29,5 @@ addSbtPlugin("pl.project13.scala" % "sbt-jmh" % "0.1.6")
 
 addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.0.0-RC1")
+
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.1.8")
```
:+1:

patriknw replied:
Ah, I didn't intend to leave it here. Do you think it should be included?

ktoso replied:
No strong opinion, it's one of these that I see people usually have in their global settings. 
Can be removed I guess, each plugin adds to load time of the build.

:+1: the use of it anyway, it's a great plugin..

drewhk replied:
Why not? If something goes wrong we can just remove it. We can make it a habit to run it every once in a while.

ktoso commented:
Nice :-) Please merge as soon as validation confirms it's happy and I'll follow up with my PR-validation-speedup

hepin1989 commented:
io.netty:netty: 3.8.0.Final -> 3.8.3.Final
just released netty 3.10.2

why not 3.10.2?play is using 3.10.x

for BC?

drewhk commented:
Yes, 2.4. aims to be bin-compat with 2.3

hepin1989 commented:
@drewhk Maybe I am asking stupidly,could I asking you why the binary compatibility is so important for two big release?I see you guys making great effect on it.
thanks.

ktoso commented:
Note to self (just realized this, will update the validator task): 
* such PR should trigger all projects to be tested - dependency changes could have dome something weird in any of the projects.

patriknw commented:
@hepin1989 good point about Netty, especially since one of our goals are to be compatible with Play 2.4, event though it is released for Akka 2.3.

I have changed to Netty 3.10.3.Final without any code changes and remoting tests pass, so I think that is a good indication that it will work.

drewhk commented:
@patriknw the issue is not as much our code, but those who happen to use the transitively included Netty AFAIK. Can we run MiMa directly on the netty artifacts?

patriknw commented:
mima between netty 3.8.0 and 3.10.3:

    abstract method getUserDefinedWritability(Int)Boolean in interface org.jboss.netty.channel.Channel does not have a correspondent in old version

    abstract method newWorker(java.util.concurrent.Executor)org.jboss.netty.channel.socket.nio.AbstractNioWorker in class org.jboss.netty.channel.socket.nio.AbstractNioWorkerPool does not have a correspondent in new version

    abstract method newWorker(java.util.concurrent.Executor)org.jboss.netty.channel.socket.nio.AbstractNioWorker in class org.jboss.netty.channel.socket.nio.AbstractNioWorkerPool does not have a correspondent in old version

    abstract method setUserDefinedWritability(Int,Boolean)Unit in interface org.jboss.netty.channel.Channel does not have a correspondent in old version

    abstract method submitWrite(org.jboss.netty.channel.ChannelHandlerContext,org.jboss.netty.channel.MessageEvent,Long,Long,Long)Unit in class org.jboss.netty.handler.traffic.AbstractTrafficShapingHandler does not have a correspondent in old version

    class org.jboss.netty.channel.StaticChannelPipeline does not have a correspondent in new version

    class org.jboss.netty.handler.codec.http.CookieDecoder is declared final in new version

    class org.jboss.netty.handler.codec.http.multipart.HttpPostRequestDecoder#IncompatibleDataDecoderException does not have a correspondent in new version

    class org.jboss.netty.handler.codec.http.websocket.DefaultWebSocketFrame does not have a correspondent in new version

    class org.jboss.netty.handler.codec.http.websocket.WebSocketFrameDecoder does not have a correspondent in new version

    class org.jboss.netty.handler.codec.http.websocket.WebSocketFrameEncoder does not have a correspondent in new version

    class org.jboss.netty.handler.codec.serialization.CompatibleObjectDecoder does not have a correspondent in new version

    class org.jboss.netty.util.internal.StackTraceSimplifier does not have a correspondent in new version

    interface org.jboss.netty.channel.ChannelPipelineCoverage does not have a correspondent in new version

    interface org.jboss.netty.handler.codec.http.websocket.WebSocketFrame does not have a correspondent in new version

    method cleanup(org.jboss.netty.channel.ChannelHandlerContext,org.jboss.netty.channel.ChannelStateEvent)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder does not have a correspondent in new version

    method compareTo(org.jboss.netty.handler.codec.http.Cookie)Int in class org.jboss.netty.handler.codec.http.DefaultCookie does not have a correspondent with same parameter signature among (java.lang.Object)Int, (org.jboss.netty.handler.codec.http.cookie.Cookie)Int

    method createWorker(java.util.concurrent.Executor)org.jboss.netty.channel.socket.nio.AbstractNioWorker in class org.jboss.netty.channel.socket.nio.NioDatagramWorkerPool does not have a correspondent in new version

    method createWorker(java.util.concurrent.Executor)org.jboss.netty.channel.socket.nio.AbstractNioWorker in class org.jboss.netty.channel.socket.nio.NioWorkerPool does not have a correspondent in new version

    method decode(org.jboss.netty.channel.ChannelHandlerContext,org.jboss.netty.channel.Channel,org.jboss.netty.buffer.ChannelBuffer)java.lang.Object in class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder does not have a correspondent in new version

    method getAggregatedText()java.lang.String in class org.jboss.netty.handler.codec.http.websocketx.ContinuationWebSocketFrame does not have a correspondent in new version

    method handleDownstream(org.jboss.netty.channel.ChannelHandlerContext,org.jboss.netty.channel.ChannelEvent)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameEncoder does not have a correspondent in new version

    method reset()Unit in class org.jboss.netty.handler.codec.spdy.SpdyHeaderBlockRawDecoder does not have a correspondent in new version

    method setAggregatedText(java.lang.String)Unit in class org.jboss.netty.handler.codec.http.websocketx.ContinuationWebSocketFrame does not have a correspondent in new version

    method setInterestOpsNow(Int)Unit in class org.jboss.netty.channel.AbstractChannel does not have a correspondent in new version

    method setInterestOpsNow(Int)Unit in class org.jboss.netty.channel.AbstractServerChannel does not have a correspondent in new version

    method this()Unit in class org.jboss.netty.handler.codec.frame.FrameDecoder was public; is inaccessible in new version

    method this(Boolean,Int,org.jboss.netty.buffer.ChannelBuffer,java.lang.String)Unit in class org.jboss.netty.handler.codec.http.websocketx.ContinuationWebSocketFrame does not have a correspondent in new version

    method this(javax.net.ssl.SSLEngine,Boolean,java.util.concurrent.Executor)Unit in class org.jboss.netty.handler.ssl.SslHandler's type has changed; was (javax.net.ssl.SSLEngine,Boolean,java.util.concurrent.Executor)Unit, is now: (javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,Boolean)Unit

    method this(javax.net.ssl.SSLEngine,java.util.concurrent.Executor)Unit in class org.jboss.netty.handler.ssl.SslHandler does not have a correspondent with same parameter signature among (javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool)Unit, (javax.net.ssl.SSLEngine,Boolean)Unit

    method this(javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,Boolean,java.util.concurrent.Executor,org.jboss.netty.util.Timer,Long)Unit in class org.jboss.netty.handler.ssl.SslHandler does not have a correspondent in new version

    method this(javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,Boolean,java.util.concurrent.Executor)Unit in class org.jboss.netty.handler.ssl.SslHandler does not have a correspondent in new version

    method this(javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,java.util.concurrent.Executor)Unit in class org.jboss.netty.handler.ssl.SslHandler's type has changed; was (javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,java.util.concurrent.Executor)Unit, is now: (javax.net.ssl.SSLEngine,org.jboss.netty.handler.ssl.SslBufferPool,Boolean)Unit

    method this(org.jboss.netty.handler.codec.spdy.SpdyVersion,Int,Int,Int)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameEncoder does not have a correspondent in new version

    method this(org.jboss.netty.handler.codec.spdy.SpdyVersion,Int,Int)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder's type has changed; was (org.jboss.netty.handler.codec.spdy.SpdyVersion,Int,Int)Unit, is now: (org.jboss.netty.handler.codec.spdy.SpdyVersion,org.jboss.netty.handler.codec.spdy.SpdyFrameDecoderDelegate,Int)Unit

    method this(org.jboss.netty.handler.codec.spdy.SpdyVersion,Int,org.jboss.netty.handler.codec.spdy.SpdyHeaderBlockDecoder)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder's type has changed; was (org.jboss.netty.handler.codec.spdy.SpdyVersion,Int,org.jboss.netty.handler.codec.spdy.SpdyHeaderBlockDecoder)Unit, is now: (org.jboss.netty.handler.codec.spdy.SpdyVersion,org.jboss.netty.handler.codec.spdy.SpdyFrameDecoderDelegate,Int)Unit

    method this(org.jboss.netty.handler.codec.spdy.SpdyVersion,org.jboss.netty.handler.codec.spdy.SpdyHeaderBlockEncoder)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameEncoder does not have a correspondent in new version

    method this(org.jboss.netty.handler.codec.spdy.SpdyVersion)Unit in class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder does not have a correspondent in new version

    method useSessionFlowControl()Boolean in class org.jboss.netty.handler.codec.spdy.SpdyVersion does not have a correspondent in new version

    the type hierarchy of class org.jboss.netty.handler.codec.spdy.SpdyFrameDecoder has changed in new version. Missing types {org.jboss.netty.handler.codec.frame.FrameDecoder,org.jboss.netty.channel.SimpleChannelUpstreamHandler}

    the type hierarchy of class org.jboss.netty.handler.codec.spdy.SpdyFrameEncoder has changed in new version. Missing types {org.jboss.netty.channel.ChannelDownstreamHandler,org.jboss.netty.channel.ChannelHandler}

plus removal of deprecated

drewhk commented:
Last time we did such an update we were strict on BC with Netty. I feel however unconfortable with lagging so much behind. Can we forfeit BC for netty for 2.4? @rkuhn ?

patriknw commented:
Rebased and added a word about the Netty dependency in the migration guide: https://github.com/akka/akka/pull/17433/files#diff-ad4d26653ea95b6bbfa0d84b90497355R27
OK?

drewhk commented:
LGTM

patriknw commented:
Refs #16695

agourlay commented:
FIY ```metrics``` has a new GroupId for artifacts post ```3.0.2```
The latest version is ```3.1.2```

see central repository https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22io.dropwizard.metrics%22%20AND%20a%3A%22metrics-core%22

patriknw commented:
@agourlay thanks for the update
I think I tried to update the metrics (or some related lib) but it required more code changes than I wanted to investigate here (we only use it for tests). I will create a separate ticket for that.
